packages/react-examples/src/react/FocusTrapZone/e2e/FocusTrapZone.e2e.ts

```diff
@@ -1,24 +1,485 @@
 
-      cy.get('#a').focus();
-      cy.focused().should('have.id', 'a');
+      // initial focus goes into the first FocusZone
+      cy.focused().should('have.text', 'fz1First');
 
-      // shift+tab to focus first bumper
+      // shift+tab to focus first bumper => wraps to second FocusZone => first button inside it
       cy.realPress(['Shift', 'Tab']);
-      cy.focused().should('have.id', 'd');
+      cy.focused().should('have.text', 'fz2First');
 
-      // tab to focus last bumper
+      // tab to focus last bumper => wraps to first FocusZone => first button inside it
       cy.realPress('Tab');
-      cy.focused().should('have.id', 'a');
+      cy.focused().should('have.text', 'fz1First');
+    });
+
+    it(
+      'can trap focus when FTZ bookmark elements are FocusZones, ' +
+        'and those elements have inner elements focused that are not the first inner element',
+      () => {
+        // This story has a FTZ containing a FocusZone (with buttons), a button, and another FocusZone (with buttons).
+        // "Bookmark" refers to the first and last elements inside the FTZ.
+        cy.loadStory(ftzStoriesTitle, 'TabWrappingFocusZoneBumpers');
+
+        // wait for first focus to finish to avoid timing issue
+        cy.focused().should('have.text', 'fz1First');
+
+        // Focus the middle button in the first FZ.
+        cy.contains('fz1First').click().realPress('ArrowRight');
+        cy.focused().should('have.text', 'fz1Mid');
+
+        // Focus the middle button in the second FZ.
+        cy.contains('fz2Mid').click().realPress('ArrowRight');
+        cy.focused().should('have.text', 'fz2Last');
+
+        // tab to focus last bumper => wraps to first FocusZone => previously focused button inside it
+        cy.realPress('Tab');
+        cy.focused().should('have.text', 'fz1Mid');
+
+        // shift+tab to focus first bumper => wraps to last FocusZone => previously focused button inside it
+        cy.realPress(['Shift', 'Tab']);
+        cy.focused().should('have.text', 'fz2Last');
+      },
+    );
+  });
+
+  describe('Tab and shift-tab when the FTZ contains 0 tabbable items', () => {
+    beforeEach(() => {
+      // This story has a FocusTrapZone containing buttons with tabIndex=-1, so they can still be
+      // clicked or programmatically focused, but aren't keyboard-focusable with tab
+      cy.loadStory(ftzStoriesTitle, 'NoTabbableItems');
+    });
+
+    it('focuses first focusable element when focusing first bumper', () => {
+      setProps({});
+
+      // wait for first focus to finish to avoid timing issue
+      cy.focused().should('have.text', 'first');
+
+      cy.contains('mid').realClick();
+      cy.focused().should('have.text', 'mid');
+
+      // shift+tab focuses the first bumper (since the buttons inside aren't keyboard-focusable)
+      //   => sends focus to first element
+      cy.realPress(['Shift', 'Tab']);
+      cy.focused().should('have.text', 'first');
```

Comment:
Maybe it'd be more informative to go to `last` instead of mid, because doing a `Shift + Tab` on `mid` with focusable buttons would also lead to `first`, whereas it would go to `mid` when doing a `Shift + Tab` on `last` with focusable buttons.